### PR TITLE
Fix 4077 metadata support

### PR DESCRIPTION
**Pull request by ismailsunni**
### What does it fix ?
* Ticket : #4077
* Funded by : DFAT
* Description : 
  - Make sure the metadata able to support field groups
  - Add group in the field

### CODE REVIEW

File: safe/definitions/field_groups.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    'fields': [
+        youth_count_field,
+        adult_count_field,
+        elderly_count_field
+    ]
+}
+
+age_ratio_group = {
+    'key': 'age_ratio_group',
+    'name': 'Age Ratio Group',
```
Comment (Gustry) [inline safe/definitions/field_groups.py]: no translation on these two names ?

File: safe/utilities/test/test_metadata.py [hunk 2]
```diff
@@ -24,6 +24,7 @@
 
 class TestMetadataUtilities(unittest.TestCase):
     """Test for Metadata Utilities module."""
+    maxDiff = None
```
Comment (Gustry) [inline safe/utilities/test/test_metadata.py]: Landscape likes when there is one blank line before and after class docstring ;-)

File: safe/definitions/field_groups.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+)
+
+__copyright__ = "Copyright 2017, The InaSAFE Project"
+__license__ = "GPL version 3"
+__email__ = "info@inasafe.org"
+__revision__ = '$Format:%H$'
+
+age_count_group = {
+    'key': 'age_count_group',
+    'name': 'Age Count Group',
```
Comment (timlinux) [inline safe/definitions/field_groups.py]: Remember Charlotte didn't like the word 'Group' too much, so for the user facing name, just call it 'Age counts' (and similar for the others).

File: safe/definitions/field_groups.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    'description': tr(
+        'The group of fields that consists of population count per age '
+        'class.'),
+    'fields': [
+        youth_count_field,
```
Comment (timlinux) [inline safe/definitions/field_groups.py]: You are going to add the other two age classes here later?

File: safe/definitions/field_groups.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+age_ratio_group = {
+    'key': 'age_ratio_group',
+    'name': 'Age Ratio Group',
+    'description': tr(
+        'The group of fields that consists of population ratio per age '
+        'class.'),
+    'fields': [
+        youth_ratio_field,
```
Comment (timlinux) [inline safe/definitions/field_groups.py]: You are going to add the other two age classes here later?

File: safe/utilities/test/test_metadata.py [hunk 6]
```diff
@@ -34,7 +35,7 @@ def test_write_iso19115_metadata(self):
         keywords = {
             'date': '26-03-2015 14:03',
             'exposure': 'structure',
-            'keyword_version': '3.2',
+            'keyword_version': '',
```
Comment (timlinux) [inline safe/utilities/test/test_metadata.py]: 4.1?

File: safe/definitions/field_groups.py (hunk 1 above)
Comment (ismailsunni) [inline safe/definitions/field_groups.py]: Ah yes, sorry. Too much c++ lately :)

File: safe/definitions/field_groups.py (hunk 3 above)
Comment (ismailsunni) [inline safe/definitions/field_groups.py]: Ok.

File: safe/definitions/field_groups.py (hunk 4 above)
Comment (ismailsunni) [inline safe/definitions/field_groups.py]: Yes, for now I will just work on prove on concept. Adding more classess or group should be easy if everything if the framework has been finished.

File: safe/definitions/field_groups.py (hunk 5 above)
Comment (ismailsunni) [inline safe/definitions/field_groups.py]: Yes, same as above

File: safe/utilities/test/test_metadata.py (hunk 6 above)
Comment (ismailsunni) [inline safe/utilities/test/test_metadata.py]: It was meant to be inasafe_keyword_version like the other two in the same file. Thanks for pointing this.

File: safe/utilities/test/test_metadata.py (hunk 2 above)
Comment (ismailsunni) [inline safe/utilities/test/test_metadata.py]: Ah yes, thanks. Class from the old time :)

**Comment by ismailsunni** [comment]
> I guess you are just making it work with the existing classes first and infant etc. are coming later?

@timlinux yes, exactly. Make it work first, then adding new classes should be simple since we don't hard code anything.

**Pull request merged**